### DISCUSSION TITLE
Allow to conditionally enable or disable the support bubble

### DISCUSSION THREAD

**Discussion by slaFFik** [spatie/laravel-support-bubble#7] [Ideas]
Use case:
The service has 2 pricing tiers, free and paid.
For free users - no support.
For paid users - email support via support bubble.

It would be nice to conditionally display/hide this bubble and deregister API endpoints based on some condition (in this case - user account level).

What do you think?

**Comment by freekmurze** [comment]
We want to keep this package as simple as possible, so likely we won't add this feature.

**Reply by slaFFik** [reply]
Okay, thanks for the reply.

I assume the only way to do that is to fork, extend your ServiceProvider class, and include the logic to render empty string for the views and not register the route. And then use the forked version in own projects.

**Reply by AlexVanderbist** [reply]
You could wrap the `<x-support-bubble />` in `@if()` blade tags to conditionally show it. Additionally you can put the routes macro in a `Route::midleware('vip-support-only')->group()` to make sure only the correct users can use it.
